app/models/user.rb

```diff
@@ -554,7 +557,7 @@ def authenticate_by_personal_access_token(token)
 
   def prepare_password
     if password.present?
-      self.password_salt = Digest::SHA1.hexdigest([Time.now.utc, rand].join)
+      self.password_salt = salt_password
```

Comment:
We are storing the salt twice for the new implementation, right? Would it be safe to assume we are using new (bcrypt) implementation for all new passwords and thus leave the salt column blank?
It would help with the `detect_implementation` as well :+1:

Reply:
I like it, let me think about it.

Reply:
Hmm you know what? As much as I would like to have the `password_salt` empty or nil, it makes the code in User model class unnecessary complex. I need to test for `password_salt` existence on multiple places as SHA1 is built around salt separated from the password. I cannot take the nice approach of defining some kind of Password model because this will not work for SHA1. I dunno, will take a look in the morning if I am able to come up with something that looks nice. I might stick with simple copy.

Reply:
if it would make the code less readable, I believe it is not necessary, we are still being able to guess the algorithm used safely.

Reply:
It's probably not worth the effort.